webgoat-lessons/xxe/src/main/java/org/owasp/webgoat/xxe/SimpleXXE.java

```diff
@@ -77,4 +85,20 @@ private boolean checkSolution(Comment comment) {
         }
         return success;
     }
+    
+    @RequestMapping(path="/xxe/tmpdir",consumes = ALL_VALUE, produces=MediaType.TEXT_PLAIN_VALUE)
+    @ResponseBody
+    public String getWebGoatHomeDirectory() {
+    	return webGoatHomeDirectory;
+    }
+    
+    @RequestMapping(path="/xxe/sampledtd",consumes = ALL_VALUE, produces=MediaType.TEXT_PLAIN_VALUE)
+    @ResponseBody
+    public String getSampleDTDFile() {
+    	return "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" + 
```

Comment:
I like the idea but I think it should be more of a hint instead of the complete dtd. (It is not really a `sampledtd` it is the complete answer. And this is the last assignment in the lesson which should not be as trivial to solve. What do you think?

Reply:
I think you are right and they still can see the value for home directory on the lesson page. So at least I could make it like: file:<<webgoat temp directory>>/XXE/secret.txt